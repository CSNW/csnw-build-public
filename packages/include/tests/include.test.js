const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const cwd = join(__dirname, 'fixtures');
const include = require('../')('nested/b.js', { cwd, baseDir: cwd });

test('pipeline', async () => {
  const process = await include.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('a.js', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
