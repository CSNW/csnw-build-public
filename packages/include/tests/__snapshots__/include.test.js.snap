// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/include/tests/fixtures",
    "contents": "var name = 'a';
",
    "cwd": "C:/dev/csnw-build/packages/include/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/include/tests/fixtures/a.js",
    "sourceMap": null,
  },
  "nested/b.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/include/tests/fixtures",
    "contents": "var name = 'b';
",
    "cwd": "C:/dev/csnw-build/packages/include/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/include/tests/fixtures/nested/b.js",
    "sourceMap": null,
  },
}
`;
